spectral_fitting/fitting_with_spectralcube.ipynb

```diff
@@ -0,0 +1,697 @@
+    "\n",
+    "## Requires ##\n",
+    "\n",
+    "```\n",
+    "pip install astropy\n",
+    "pip install spectral-cube\n",
+    "pip install radio-beam\n",
+    "pip install dask\n",
+    "pip install aplpy\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from spectral_cube import SpectralCube, Projection\n",
+    "import astropy.units as u\n",
+    "from astropy.io import fits\n",
+    "from astropy.modeling import models, fitting\n",
+    "from astropy.convolution import convolve_models\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Fitting models to both spatial maps and spectra are amongst the most common operations involving spectral-line data cubes.\n",
+    "\n",
+    "In this tutorial, we explore how to fit models to a `SpectralCube` using `astropy.modeling`. To show this, we will use an ALMA mosaic observed with the ACA, or 7-m array. This mosaic is one component of an ALMA project to map CO(2-1) in M33 (Project ID 2019.1.01182.S).\n",
+    "\n",
+    "To keep the data volume small, we will use a 10 MB cutout of the full mosaic for this tutorial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy.utils.data import download_file\n",
+    "\n",
+    "datafile = download_file(\n",
+    "    'https://zenodo.org/record/4021108/files/M33_Brick1Tile1_12CO21_0p7kms.image.pbcor_K.cutout.fits',\n",
+    "    cache=True, show_progress=True)\n",
+    "\n",
+    "# Note that the datafile will not end in \".fits\". Because of that, we need to specify the format\n",
+    "# When the file name end in \"fits\", `format` will not need to be specified \n",
+    "cube = SpectralCube.read(datafile, format='fits')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The basic cube properties are:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": "SpectralCube with shape=(300, 64, 64) and unit=K:\n n_x:     64  type_x: RA---SIN  unit_x: deg    range:    23.408277 deg:   23.428647 deg\n n_y:     64  type_y: DEC--SIN  unit_y: deg    range:    30.750561 deg:   30.768065 deg\n n_s:    300  type_s: VRAD      unit_s: m / s  range:  -280000.000 m / s:  -79670.000 m / s"
+     },
+     "metadata": {},
+     "execution_count": 2
+    }
+   ],
+   "source": [
+    "cube"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the spectral unit is in m/s from the FITS file. It will be convenient to instead use km/s, so we will convert the cube's spectral unit:"
```

Comment:
Note that the spectral unit -> Note that the unit of the spectral axis (just for clarity)

Reply:
Fixed